pycqed/instrument_drivers/meta_instrument/device.py

```diff
@@ -0,0 +1,991 @@
+        MC.set_sweep_function(hard_sweep_func)
+        MC.set_sweep_points(hard_sweep_points if classified else
+                            hard_sweep_points * max(qb.acq_shots() for qb in qubits))
+
+        MC.set_sweep_function_2D(awg_swf.SegmentSoftSweep(
+            hard_sweep_func, sequences, 'Nr. Seeds', ''))
+        MC.set_sweep_points_2D(soft_sweep_points)
+        det_get_values_kws = {'classified': classified,
+                              'correlated': correlated,
+                              'thresholded': thresholded,
+                              'averaged': averaged}
+        if classified:
+            det_type = 'int_avg_classif_det'
+            nr_shots = max(qb.acq_averages() for qb in qubits)
+        else:
+            det_type = 'int_log_det'
+            nr_shots = max(qb.acq_shots() for qb in qubits)
+        det_func = mqm.get_multiplexed_readout_detector_functions(
+            qubits, nr_averages=max(qb.acq_averages() for qb in qubits),
+            nr_shots=nr_shots, det_get_values_kws=det_get_values_kws)[det_type]
+        MC.set_detector_function(det_func)
+
+        # create sweep points
+        sp = SweepPoints('nr_seeds', np.arange(nr_seeds), '', 'Nr. Seeds')
+        sp.add_sweep_dimension()
+        sp.add_sweep_parameter('cliffords', cliffords, '',
+                               'Number of applied Cliffords, $m$')
+
+        # create analysis pipeline object
+        meas_obj_value_names_map = mqm.get_meas_obj_value_names_map(qubits,
+                                                                    det_func)
+        mobj_names = list(meas_obj_value_names_map)
+        pp = ProcessingPipeline(meas_obj_value_names_map)
+        for i, mobjn in enumerate(mobj_names):
+            pp.add_node(
+                'average_data', keys_in='raw',
+                shape=(len(cliffords), nr_seeds), meas_obj_names=[mobjn])
+            pp.add_node(
+                'get_std_deviation', keys_in='raw',
+                shape=(len(cliffords), nr_seeds), meas_obj_names=[mobjn])
+            pp.add_node(
+                'SingleQubitRBAnalysis', keys_in='previous average_data',
+                std_keys='previous get_std_deviation',
+                meas_obj_names=[mobjn], plot_T1_lim=False, d=4)
+        # create experimental metadata
+        exp_metadata = {'preparation_params': prep_params,
+                        'cal_points': repr(cp),
+                        'sweep_points': sp,
+                        'meas_obj_sweep_points_map':
+                            {qbn: ['nr_seeds', 'cliffords'] for
+                             qbn in mobj_names},
+                        'meas_obj_value_names_map': meas_obj_value_names_map,
+                        'processing_pipe': pp}
+        MC.run_2D(name=label, exp_metadata=exp_metadata)
+
+        if analyze_RB:
+            pla.PipelineDataAnalysis()
+
+    def measure_chevron(self, qbc, qbt, hard_sweep_params, soft_sweep_params,
```

Comment:
Merging all these functions in the master will require extra care, because some of those have been modified in the master branch which you branched off a long time ago, but they won't cause any merge conflict since you are writing them in a new file. To avoid introducing again all the bugs that have been fixed, the last version of these functions currently in the master (valid also for measure cphase, etc.) should be taken, which is currently not the case.

Reply:
Thank you, Nathan, for pointing that out. It raises another question: there is a branch enh/cphase_dynphase with many further improvements to cphase and dynphase measurements. If Michael has to copy new versions of the functions anyways, would it make sense to first process a pull request from the other branch to have the most recent versions in the master, and then copy (and adapt) the most recent versions afterwards? Otherwise we do the same work twice.

And why are the functions not removed from mqm if they are now here? Is that intended for backward compatibility? If yes, the functions in mqm should be replaced by wrappers that create a device object and call the new functions. Otherwise, if we do not need this backward compatibility, the functions in mqm should be removed.

Reply:
You mean specifically for the methods in the multi_qubit_module? It should not require too much work to compare them with the ones in the master branch.

Reply:
Yes, I will then wait with adapting the routines until the other PR is finished. Who is responsible for that one?

Concerning the mqm routines. I think we should delete them, as we should refrain from calling the method outside our device instance, which contains all device parameters.

Reply:
As discussed with @chellings I updated the methods in the device class to use the current version in the multi qubit module of the master branch. 

I also deleted the version in the multiqubit module and have errors raised when the method is being called.